### PR TITLE
Asset filtering for eval and training

### PR DESCRIPTION
**Pull request by antonioloison**
Add asset filter similar to Kili SDK assets method that can filter on any of the following fields:
- "asset_id_in"
- "consensus_mark_gt"
- "consensus_mark_lt"
- "external_id_contains",
- "honeypot_mark_gt",
- "honeypot_mark_lt",
- "label_author_in",
- "label_created_at",
- "label_created_at_gt",
- "label_created_at_lt",
- "label_category_search",
- "label_type_in",
- "metadata_where",
- "updated_at_gte",
- "updated_at_lte"

### CODE REVIEW

File: commands/common_args.py [hunk 1]
```diff
@@ -10,6 +10,15 @@
 DEFAULT_BATCH_SIZE = 8
 
 
+def asset_filter_loader(json_string):
```
Comment (PierreLeveau) [inline commands/common_args.py:13]: ` Optional[str]`

File: kiliautoml/utils/helpers.py [hunk 2]
```diff
@@ -78,8 +79,34 @@ def get_asset_memoized(
     total: Optional[int],
     skip: int,
     status_in: Optional[List[AssetStatusT]] = None,
+    asset_filter: Optional[AssetFilterArgsT] = None,
 ) -> List[Any]:
     kili.assets = backoff.on_exception(backoff.expo, exception=Exception, max_tries=3)(kili.assets)
+
+    additional_filtering_arguments: Dict[str, Any] = {}
+    if asset_filter is not None:
```
Comment (PierreLeveau) [inline kiliautoml/utils/helpers.py:87]: using a dict comprehension would be a bit lighter: https://stackoverflow.com/questions/3420122/filter-dict-to-contain-only-certain-keys

File: kiliautoml/models/_hugging_face_text_classification_model.py [hunk 3]
```diff
@@ -78,6 +78,11 @@ def train(
 
         path_dataset = os.path.join(PathHF.dataset_dir(model_repository_dir), "data.json")
         logger.info(f"Downloading data to {path_dataset}")
+        if not clear_dataset_cache:
```
Comment (PierreLeveau) [inline kiliautoml/models/_hugging_face_text_classification_model.py:81]: why only for text classification? that's probably the least used task.

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: kiliautoml/utils/helpers.py (hunk 2 above)
Comment (antonioloison) [reply kiliautoml/utils/helpers.py:87]: with the if else condition to log the warning, it is complicated to use dict comprehension; it feels like code is less readable, don't you think?

File: kiliautoml/models/_hugging_face_text_classification_model.py (hunk 3 above)
Comment (antonioloison) [reply kiliautoml/models/_hugging_face_text_classification_model.py:81]: cache is not implemented in NER, I created a branch to implement it for NER but I think we should make another ticket for that

File: kiliautoml/utils/helpers.py (hunk 2 above)
Comment (PierreLeveau) [reply kiliautoml/utils/helpers.py:87]: ok good point

File: kiliautoml/models/_hugging_face_text_classification_model.py (hunk 3 above)
Comment (PierreLeveau) [reply kiliautoml/models/_hugging_face_text_classification_model.py:81]: 👍

**Review by PierreLeveau — APPROVED** [review]

**Pull request merged**